api/app/Repositories/User/UserRepository.php

```diff
@@ -71,8 +71,14 @@ public function findOneByCriteria(EloquentCriterion ...$criteria): User
         return $query->first();
     }
 
-    public function getGoogleCalendarTokenById(int $id): array
+    public function getGoogleCalendarTokenById(int $id): ?array
     {
-        return User::findOrFail($id)->googleAccounts[0]->token;
+        $user = User::findOrFail($id);
+
+        if (count($user->googleAccounts)) {
```

Comment:
I think you can check without `count`
`if ($user->googleAccounts)`

Reply:
Then we'll have the same error cause googleAccounts might be an empty collection and index 0 will be unavailable;